### PR TITLE
linkserver: nxp: add support for frdm-k64f

### PR DESCRIPTION
**Pull request by yvanderv**
- board.cmake is augmented for LinkServer
 - the board's doc file is updated as well

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
Hi, thanks for getting this enabled! A few comments here:
- we should not remove the `pyocd` installation instructions in this PR- that host tool is still the default for this board, and we should make sure users are aware of how to set it up when they look at the board documentation page
- the commit title should be something like the following `boards: frdm_k64f: enable linkserver support`

File: boards/arm/frdm_k64f/board.cmake [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 board_runner_args(jlink "--device=MK64FN1M0xxx12")
+board_runner_args(linkserver  "--device=MK64FN1M0xxx12:FRDM-K64F")
+
 board_runner_args(pyocd "--target=k64f")
 
 include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/canopen.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/linkserver.board.cmake)
```
Comment (mmahadevan108) [inline boards/arm/frdm_k64f/board.cmake:12]: Recommend this be moved to be first so LinkServer is the default.

**Comment by mmahadevan108** [comment]
Please update the commit message to indicate that linkserver is the new default.

**Review by mmahadevan108 — DISMISSED** [review]

File: boards/arm/frdm_k64f/doc/index.rst [hunk 2]
```diff
@@ -241,16 +241,29 @@ configured by default to use the :ref:`opensda-daplink-onboard-debug-probe`.
...
 
    .. group-tab:: OpenSDA DAPLink Onboard (Recommended)
 
-        Install the :ref:`pyocd-debug-host-tools` and make sure they are in your search
-        path.
+	Install the :ref:`linkserver-debug-host-tools` and make sure they are in your
+	search path.  LinkServer works with the default CMSIS-DAP firmware included in
+	the on-board debugger.
+
+        Linkserver is the default for this board, ``west flash`` and ``west debug`` will
```
Comment (danieldegrasse) [inline boards/arm/frdm_k64f/doc/index.rst:254]: When I test this PR locally, Linkserver does not appear to be the defaut:

```
(.venv) ~/zephyr/zephyrproject/zephyr:(177314f17a...) ❯ west flash
-- west flash: rebuilding
[1/1] cd /home/danieldegrasse/zephyr/zephyrproject...ephyr/zephyrproject/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner pyocd
WARNING: runners.pyocd: hex file (None) does not exist; falling back on .bin (/home/danieldegrasse/zephyr/zephyrproject/zephyr/build/zephyr/zephyr.bin). Consider enabling CONFIG_BUILD_OUTPUT_HEX.
-- runners.pyocd: Flashing file: /home/danieldegrasse/zephyr/zephyrproject/zephyr/build/zephyr/zephyr.bin
0001551 I Loading /home/danieldegrasse/zephyr/zephyrproject/zephyr/build/zephyr/zephyr.bin [load_cmd]
[==================================================] 100%
0002650 I Erased 0 bytes (0 sectors), programmed 0 bytes (0 pages), skipped 20480 bytes (5 pages) at 18.21 kB/s [loader]
(.venv) ~/zephyr/zephyrproject/zephyr:(177314f17a...) 
```
versus: 
```
west flash -r linkserver
-- west flash: rebuilding
[1/1] cd /home/danieldegrasse/zephyr/zephyrproject...ephyr/zephyrproject/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner linkserver
-- runners.linkserver: LinkServer: /usr/local/LinkServer/LinkServer, version v1.3.15
INFO: Exact match for MK64FN1M0xxx12:FRDM-K64F found
INFO: Selected device MK64FN1M0xxx12:FRDM-K64F
INFO: Getting available probes
INFO: Selecting probe by index
INFO: Selected probe #1 02400b0128884e45004400106bf000478021000097969900 (DAPLink CMSIS-DAP
```

**Review by danieldegrasse — COMMENTED** [review]
We need to update the `linkserver.board.cmake` file to include `board_set_flasher_ifnset` directives, like other runner cmake files do: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/common/jlink.board.cmake#L3. Otherwise, linkserver will not actually become the default

File: boards/arm/frdm_k64f/doc/index.rst (hunk 2 above)
Comment (DerekSnell) [inline boards/arm/frdm_k64f/doc/index.rst:254]: When LinkServer is the default runner, this doc needs some additional changes.  On the tab "OpenSDA JLink Onboard", change this sentence:
> Add the arguments ... when you invoke west build to override the default runner from pyOCD to J-Link.

to
> Add the arguments ... when you invoke west build to override the default runner to J-Link.

**Review by DerekSnell — CHANGES_REQUESTED** [review]
Thanks @yvanderv for this update.
But the doc page needs some updates, thanks

**Comment by yvanderv** [comment]
> We need to update the `linkserver.board.cmake` file to include `board_set_flasher_ifnset` directives, like other runner cmake files do: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/common/jlink.board.cmake#L3. Otherwise, linkserver will not actually become the default

Sure, will submit a separate PR.

**Comment by danieldegrasse** [comment]
> Sure, will submit a separate PR.

Ok, and I assume that this PR would then be dependent on that one, since we want to move linkserver to be the default on this board?

**Review by DerekSnell — APPROVED** [review]
Thanks, @yvanderv !
LinkServer is now the default runner.

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**